### PR TITLE
feat(spanner/spannertest): support AVG aggregation function

### LINKED ISSUE #3285: spannertest: Support AVG
**Is your feature request related to a problem? Please describe.**
It's better if `spannertest` supports [`AVG`](https://cloud.google.com/spanner/docs/functions-and-operators#avg), one of the aggregated functions provided by Spanner. 

**Describe the solution you'd like**
I feel it's not so difficult because the processes should be similar with "SUM" and "COUNT", which are already supported.

### PR DESCRIPTION
**Pull request by sryoya**
Fixes #3285 

Sorry for creating PR without waiting for the response to the above issue 😅

### CODE REVIEW

File: spanner/spannertest/integration_test.go [hunk 1]
```diff
@@ -889,6 +889,42 @@ func TestIntegration_ReadsAndQueries(t *testing.T) {
 				{int64(1), int64(25)}, // Jack(ID=1, Tenure=10), Sam(ID=3, Tenure=9), George(ID=5, Tenure=6)
 			},
 		},
+		{
```
Comment (dsymonds) [inline spanner/spannertest/integration_test.go:892]: I think there's too many test cases here. This doesn't need to exhaustively test, just give it a quick exercise. Maybe just grab one of them from https://cloud.google.com/spanner/docs/aggregate_functions#avg.

File: spanner/spannertest/funcs.go [hunk 2]
```diff
@@ -109,6 +109,48 @@ var aggregateFuncs = map[string]aggregateFunc{
...
+					}
+					sum += v.(int64)
+					n++
+				}
+				if n == 0 {
+					// "Returns NULL if the input contains only NULLs".
+					return nil, typ, nil
+				}
+				// floating point handling is non-deterministic
+				// type of the result from INT64 input is Float64
```
Comment (dsymonds) [inline spanner/spannertest/funcs.go:132]: I don't think these comments are useful here.

File: spanner/spansql/keywords.go [hunk 3]
```diff
@@ -135,6 +135,7 @@ var funcs = map[string]bool{
 	"MAX":       true,
 	"MIN":       true,
 	"SUM":       true,
+	"AVG":       true,
```
Comment (dsymonds) [inline spanner/spansql/keywords.go:138]: keep these in alphabetical order.

File: spanner/spansql/parser_test.go [hunk 4]
```diff
@@ -111,6 +111,20 @@ func TestParseQuery(t *testing.T) {
 				},
 			},
 		},
+		{`SELECT AVG(PointsScored) AS total_points, FirstName, LastName AS surname FROM PlayerStats GROUP BY FirstName, LastName`,
```
Comment (dsymonds) [inline spanner/spansql/parser_test.go:114]: this test case is really not needed.

**Review by dsymonds — COMMENTED** [review]
Looks good overall, but just needs to be pared back.

File: spanner/spansql/keywords.go (hunk 3 above)
Comment (sryoya) [reply spanner/spansql/keywords.go]: I didn't notice the order rule, thanks.
https://github.com/googleapis/google-cloud-go/pull/3286/commits/64453991b15679ec55ac385d6a943f3bf999b2b2

File: spanner/spannertest/funcs.go (hunk 2 above)
Comment (sryoya) [reply spanner/spannertest/funcs.go]: Ok. As I just brought them from the official doc, we may not need to keep it in the code.
https://github.com/googleapis/google-cloud-go/pull/3286/commits/d9b8efb9a03562c1638653fe74fca69275fb9ce3

File: spanner/spannertest/integration_test.go (hunk 1 above)
Comment (sryoya) [reply spanner/spannertest/integration_test.go]: Sorry, I didn't understand the test policy about how exhaustively tests should be implemented. I kept only one simple test and removed the others.

Btw, I tried to use either of the sample SQLs written in the official doc, but they don't seem to work correctly. 
I raised that as another issue here.
https://github.com/googleapis/google-cloud-go/issues/3296

So as I said in the above, I kept one original test case using the prepared data in this test func for the time being.

File: spanner/spansql/parser_test.go (hunk 4 above)
Comment (sryoya) [reply spanner/spansql/parser_test.go:114]: Removed.
https://github.com/googleapis/google-cloud-go/pull/3286/commits/e514c20b4199458ab35272eebc11cf0bd0f05efb

**Comment by sryoya** [comment]
@dsymonds Thanks for your review. I updated the points you commented. Can you check them again?

File: spanner/spannertest/integration_test.go (hunk 1 above)
Comment (dsymonds) [reply spanner/spannertest/integration_test.go:892]: That's fine, but you can still use that example, just slightly modified. There's other test cases here like that. We don't want to tie too many test cases to the built-up test data, as if that ever changes then there's lots of tests that need adjusting; for something like `AVG` it should suffice to see it exercised in an array provided as a param.

Comment (sryoya) [reply spanner/spannertest/integration_test.go:892]: Though I added the related comment in #3296 , I cannot make the SQL work as it is in the official doc.

But,
> We don't want to tie too many test cases to the built-up test data, as if that ever changes then there's lots of tests that need adjusting; for something like AVG it should suffice to see it exercised in an array provided as a param.

I completely agree with that. Actually, I feel that when I was implementing. So, I want to somehow create the test that doesn't depend on the built-up test data.

Comment (sryoya) [reply spanner/spannertest/integration_test.go:892]: I tried to work out the case that can test "AVG" without the built-up data, but, unfortunately, I could not get that from the currently supported SQLs. 
Let me know if you have any.

Comment (dsymonds) [reply spanner/spannertest/integration_test.go:892]: Sure, alright. Let's just use the existing data for this.

**Review by dsymonds — DISMISSED** [review]
LGTM

**Review by dsymonds — APPROVED** [review]

**Pull request merged**